api/provider/aws/deploy_create.go

```diff
@@ -0,0 +1,136 @@
+	if err != nil {
+		return nil, err
+	}
+
+	bytes, err := json.Marshal(taskDefinitionOutput)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to extract deploy file: %s", err.Error())
+	}
+
+	deploy := &models.Deploy{
+		// TODO: we previously removed prefix from deployID ala below. Is this
+		// no longer needed?
+		// func removePrefix(id string) string {
+		// 	return strings.TrimPrefix(id, PREFIX)
+		// }
+		DeployID:   deployID,
+		Version:    d.getDeployRevision(deployID),
+		DeployFile: bytes,
+	}
+
+	if err := d.TagStore.Insert(models.Tag{EntityID: deploy.DeployID, EntityType: "deploy", Key: "name", Value: req.DeployName}); err != nil {
+		return deploy, err
+	}
+
+	if err := d.TagStore.Insert(models.Tag{EntityID: deploy.DeployID, EntityType: "deploy", Key: "version", Value: deploy.Version}); err != nil {
+		return deploy, err
+	}
+
+	return deploy, nil
+}
+
+func (d *DeployProvider) createTaskDefinition(taskDefinitionRequest *ecs.TaskDefinition) (*ecs.TaskDefinition, error) {
+
+	input := &ecs.RegisterTaskDefinitionInput{}
+	input.SetFamily(aws.StringValue(taskDefinitionRequest.Family))
+	input.SetTaskRoleArn(aws.StringValue(taskDefinitionRequest.TaskRoleArn))
+	input.SetNetworkMode(aws.StringValue(taskDefinitionRequest.NetworkMode))
+	input.SetContainerDefinitions(taskDefinitionRequest.ContainerDefinitions)
+	input.SetVolumes(taskDefinitionRequest.Volumes)
+	input.SetPlacementConstraints(taskDefinitionRequest.PlacementConstraints)
+
+	if err := input.Validate(); err != nil {
+		return nil, err
+	}
+
+	_, err := d.AWS.ECS.RegisterTaskDefinition(input)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: return original request or RegisterTaskDefinition() input?
+	return taskDefinitionRequest, nil
+}
+
+func (d *DeployProvider) renderTaskDefinition(body []byte, familyName string) (*ecs.TaskDefinition, error) {
+	var taskDefinition *ecs.TaskDefinition
+
+	if err := json.Unmarshal(body, &taskDefinition); err != nil {
+		return nil, fmt.Errorf("Failed to decode deploy: %s", err.Error())
+	}
+
+	if len(taskDefinition.ContainerDefinitions) == 0 {
+		return nil, fmt.Errorf("Deploy must have at least one container definition")
+	}
+
+	if taskDefinition.Family != nil && taskDefinition.Family != aws.String(familyName) {
+		return nil, fmt.Errorf("Custom family names are currently unsupported in Layer0")
+	}
+
+	for _, container := range taskDefinition.ContainerDefinitions {
+		if container.LogConfiguration == nil {
+			container.LogConfiguration = &ecs.LogConfiguration{
+				LogDriver: aws.String("awslogs"),
+				Options: map[string]*string{
+					"awslogs-group":         aws.String(fmt.Sprintf("l0-%s", d.Config.Instance())),
+					"awslogs-region":        aws.String(d.Config.Region()),
+					"awslogs-stream-prefix": aws.String("l0"),
+				},
+			}
+		}
+	}
+
+	return taskDefinition, nil
+}
+
+// TODO: remove this if possible
+func (d *DeployProvider) getDeployRevision(id string) string {
```

Comment:
remove - I think we should track deploy versions solely through tags

Reply:
Sure, but we get the value to place in the tag partially via this function. I'm fine with deleting this and putting the logic in Create() if need be, but it's just ugly

Reply:
The `version` should be taken from the `Revision` field of the [ecs.TaskDefinition](https://github.com/aws/aws-sdk-go/blob/master/service/ecs/api.go#L8494)